### PR TITLE
Split CI into different workflows

### PR DESCRIPTION
**Pull request by RemakingEden**
For scalability and readability I feel it would be easier to break our CI pipeline into distinct workflows. This is not always straightforward on Github as anything that needs to share artefacts does need to be together in a workflow however I think im getting close to something more useable.

**Description**
Split the pipeline into multiple workflows, also make the cache work and pass artefacts to different steps to harden its use for scalability.

**Status**
Ready

**Types of changes**

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "yaml.schemas": {
```
Comment (mocsharp) [inline .vscode/settings.json:2]: do we want to include this file?

**Review by mocsharp — APPROVED** [review]
LGTM

**Review by Dusty-Meg — APPROVED** [review]
Looks good.

File: .vscode/settings.json (hunk 1 above)
Comment (RemakingEden) [reply .vscode/settings.json:2]: I was unsure about this but looking at it now it isnt up to date. I'm going to take it out. Thanks @mocsharp

**Comment by RemakingEden** [comment]
Had to force push to add sign off for the DCO bot

**Pull request merged**